app/models/course_instance.rb

```diff
@@ -18,6 +18,10 @@ class CourseInstance < ActiveRecord::Base
     .unenrolled(user)
   }
 
+  def schedule
+    meetings.map(&:schedule).join('; ') || 'TBA'
```

Comment:
That one liner you mentioned also doesn't work because an empty array is still truthy.